persist/sqlite/v2consensus_refactored_test.go

```diff
@@ -115,3 +117,744 @@ func TestV2Block(t *testing.T) {
 
 	checkBlocks(1)
 }
+
+func TestV2SiacoinOutput(t *testing.T) {
+	pk1 := types.GeneratePrivateKey()
+	uc1 := types.StandardUnlockConditions(pk1.PublicKey())
+	addr1 := uc1.UnlockHash()
+	addr1Policy := types.SpendPolicy{Type: types.PolicyTypeUnlockConditions(uc1)}
+
+	pk2 := types.GeneratePrivateKey()
+	uc2 := types.StandardUnlockConditions(pk2.PublicKey())
+	addr2 := uc2.UnlockHash()
+
+	n := newTestChain(t, true, func(network *consensus.Network, genesisBlock types.Block) {
+		genesisBlock.Transactions[0].SiacoinOutputs[0].Address = addr1
+	})
+	scID := n.genesis().Transactions[0].SiacoinOutputID(0)
+
+	// genesis output should be unspent
+	// so spentIndex = nil
+	n.assertSCE(t, scID, nil, n.genesis().Transactions[0].SiacoinOutputs[0])
+
+	txn1 := types.V2Transaction{
+		SiacoinInputs: []types.V2SiacoinInput{{
+			Parent:          getSCE(t, n.db, scID),
+			SatisfiedPolicy: types.SatisfiedPolicy{Policy: addr1Policy},
+		}},
+		SiacoinOutputs: []types.SiacoinOutput{{
+			Address: addr2,
+			Value:   n.genesis().Transactions[0].SiacoinOutputs[0].Value,
+		}},
+	}
+	testutil.SignV2Transaction(n.tipState(), pk1, &txn1)
+
+	n.mineV2Transactions(t, txn1)
+
+	n.assertV2Transactions(t, txn1)
+
+	// genesis output should be spent
+	tip := n.tipState().Index
+	n.assertSCE(t, scID, &tip, n.genesis().Transactions[0].SiacoinOutputs[0])
```

Comment:
I would extract `n.genesis().Transactions[0].SiacoinOutputs[0]` and others into a variable. I'll approve the PR it's obviously a `nit` but it makes the test so much lighter and easier to read through.